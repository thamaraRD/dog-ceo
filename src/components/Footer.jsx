import PropTypes from 'prop-types';
import { footer } from "../assets/texts";

export const Footer = () => {
  return (
    <div className="fixed-bottom footer">
    <div className="row">
        <div className="col-md-12 p-0 mt-6">
            <p className=" bg-danger text-dark text-center p-0
            m-0">{footer.FOOTER_DESCRIPTION}</p>
        </div>
    </div>
</div>
  )
};

Footer.propTypes = {
  footer: PropTypes.string.isRequired,
  className: PropTypes.string,
};

Footer.defaultProps = {
  footer: 'Hecho por: Thamara Ramos',
}
