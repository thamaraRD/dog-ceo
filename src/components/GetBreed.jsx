/* eslint-disable no-unused-expressions */
import { useCallback, useEffect } from "react";
import { getBreed } from "../helpers/getAllOfBreeds";

export const GetBreed = ({ loadBreed }) => {
 

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const loadBreedHandle = useCallback(async(loadBreed) =>{
    const data = await getBreed();
    loadBreed(data)
  })
  
  useEffect(() => {
    loadBreedHandle(loadBreed)
  }, [loadBreedHandle])
  

  return(
    <div>
      
    </div>
  )
};
