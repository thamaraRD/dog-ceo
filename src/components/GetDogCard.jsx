import { useEffect, useState } from "react";
import { getBreed } from "../helpers/getAllOfBreeds";


const baseUrl = 'https://dog.ceo/api/';

const getList = async () => {

  const url = `${baseUrl}breeds/list/all`
  const resp = await fetch(url);
  const data = await resp.json();
  const breed = Object.keys(data.message);
  console.log('BREED', breed);
  
  const list = breed.map( breed => breed )
// console.log('list', list);
return list;
}
export const GetDogCard = ({breed}) => {
 
  //   const [dataBreed, setDataBreed] = useState([]);
  
  // useEffect(() => {
    // getBreed(breed).then(data => {setDataBreed(data)})
    // }, [])
 
  useEffect(() => {
   
    getList();
    console.log('getList()', getList());
  }, [])
  

  return (
    <>
   
    <h3>{breed}</h3>
    <select>
      <option>Selecciona la raza</option>
      {Object.keys(breed).map((breed, index) =>
      <option key={index}>{breed}</option>
      ).join()}
    </select>
    
    {/* {breed.map((breed, index) =>
      <p key={index}>{breed}</p>
      )} */}
    
  
    </>
  )
};

