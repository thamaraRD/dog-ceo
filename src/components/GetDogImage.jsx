import PropTypes from "prop-types";
import { useEffect, useState } from "react";
import { description } from "../assets/texts";
import { getImageBreed } from "../helpers/getAllOfBreeds";

export const GetDogImage = ({ dogImage = [] }) => {
  const [imageBreed, setImageBreed] = useState([]);
  const [dogDataImages, setDogDataImages] = useState([]);
  const [loader, setLoader] = useState(false);

  useEffect(() => {
    dogImage.map((dog) => {
      getImageBreed(dog).then((data) => {
        setImageBreed(data.splice(0, 6));
        setLoader(true);
      });
    });
  }, [dogImage]);

  useEffect(() => {
    setDogDataImages([ ...imageBreed]);
  }, [imageBreed]);

  return (
      <>
        { imageBreed.length === 0 ? (
          <div className="container mt-5 text-center">
            <h4 className="text-danger">{description.INFO_DESCRIPTION}</h4>
          </div>
        ) : (
          <div className="container-css">
        <div className="card-grid">
          {dogDataImages.map((image, index) => (
            <img className="photo-dog" key={index} src={image} alt="pretty doggui" />
          ))}
        </div>
      </div>
          )}
      </>
      )
};

GetDogImage.prototypes = {
  className: PropTypes.string,
  image: PropTypes.string
};
