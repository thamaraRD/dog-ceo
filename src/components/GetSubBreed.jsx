
export const GetSubBreed = () => {
  return (
    <input 
    type='text'
    placeholder="Â¿TendrÃ¡ Alguna sub raza? ... Â¡EncuÃ©ntrala!"
    />
  )
}
