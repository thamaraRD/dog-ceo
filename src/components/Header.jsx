import PropTypes  from "prop-types";
import Container from "react-bootstrap/Container";
import Navbar from "react-bootstrap/Navbar";
import { title, description } from "../assets/texts";
import image from '../assets/images/dog-navbar.png';
import dogFace from '../assets/images/dog-face.png'

export const Header = () => {
  return (
    <Navbar>
      <Container >
        <Navbar.Brand >
          <h1 className="redLetter">{title.TITLE}<img src={image} alt='little dog' /></h1>
        </Navbar.Brand>
        <Navbar.Toggle />
        <Navbar.Collapse className="justify-content-end">
          <Navbar.Text>
            <p className="redLetter">{description.INTRO_DESCRIPTION}<img src={dogFace} alt="dog face" /></p>
          </Navbar.Text>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

Header.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    className: PropTypes.string,
    image: PropTypes.string,
};

Header.defaultProps = {
  title: 'DOGS',
  description: '',
  className: '',
}
