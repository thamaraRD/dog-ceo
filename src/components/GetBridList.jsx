import PropTypes  from "prop-types";
import { useEffect, useState } from "react";
import { Select } from "antd";
import { getBreed } from "../helpers/getAllOfBreeds";
import "antd/dist/antd.css";
import { GetDogImage } from "./GetDogImage";

export const GetBreedList = () => {
  const [breedDog, setBreedDog] = useState([]);
  const [dogImage, setDogImage] = useState([]);
  
  useEffect(() => {
    getBreed().then((data) => {
      setBreedDog(data);
    });
  }, []);

  const handleChange = (value) => {

    setDogImage(value);
  };

  const { Option } = Select;

  return (
    <>
      <Select className="selectList"
        placeholder="Seleccione una raza o escrÃ­bala ..."
        size="large"
        showSearch
        optionFilterProp="children"
        mode="multiple"
        onChange={handleChange}
      >
        {breedDog.map((breed, index) => (
          <Option key={index} value={breed}>
            {breed}
          </Option>
        ))}
      </Select>

      <GetDogImage dogImage={dogImage} />
    </>
  );
};

GetBreedList.prototypes = {
  placeholder: PropTypes.string,
  className: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.oneOfType([PropTypes.string]),
  size: PropTypes.string,
  mode: PropTypes.string.isRequired,

};

GetBreedList.defaultProps = {
  mode: 'multiple'
};
