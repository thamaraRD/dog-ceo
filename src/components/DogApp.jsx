import { Header } from "./Header";
import { Footer } from "./Footer";
import { GetBreedList } from "./GetBridList";

export const DogApp = () => {

  return (
    <>
      <Header />
      <GetBreedList />
      <Footer />
    </>
  );
};

