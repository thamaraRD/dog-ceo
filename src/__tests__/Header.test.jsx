import { render } from "@testing-library/react";
import { Header } from "../components/Header";


describe('test on <Header />', () => { 

  
  test('should show the title in <h1> without spaces', () => { 
    
    const title = 'DOGS';
    const titleOnly = title.trim();
    
    expect( title ).toBe( titleOnly );
    
  });
  
  test('should exist className', () => {
    
    const className = 'redLetter';
    
    expect(className).toBeTruthy();
    
  });
  
  test('should match the snapshot', () => { 

    const { container } = render( <Header />);

    expect( container ).toMatchSnapshot();

   });
 });