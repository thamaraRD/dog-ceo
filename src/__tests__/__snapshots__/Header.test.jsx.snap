// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <Header /> should match the snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-light"
  >
    <div
      class="container"
    >
      <span
        class="navbar-brand"
      >
        <h1
          class="redLetter"
        >
          DOGS
          <img
            alt="little dog"
            src="dog-navbar.png"
          />
        </h1>
      </span>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="justify-content-end navbar-collapse collapse"
      >
        <span
          class="navbar-text"
        >
          <p
            class="redLetter"
          >
            Fotos de Perritos Lindos
            <img
              alt="dog face"
              src="dog-face.png"
            />
          </p>
        </span>
      </div>
    </div>
  </nav>
</div>
`;
